util/db/repository.go

```diff
@@ -3,29 +3,25 @@ package db
 import (
 	"fmt"
 	"hash/fnv"
-	"strings"
+
+	log "github.com/sirupsen/logrus"
 
 	"golang.org/x/net/context"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-	apiv1 "k8s.io/api/core/v1"
-	apierr "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	log "github.com/sirupsen/logrus"
 
-	"github.com/argoproj/argo-cd/v2/common"
 	appsv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
-	"github.com/argoproj/argo-cd/v2/util/errors"
-	"github.com/argoproj/argo-cd/v2/util/git"
-	"github.com/argoproj/argo-cd/v2/util/settings"
 )
 
 const (
 	// Prefix to use for naming repository secrets
 	repoSecretPrefix = "repo"
+	// Prefix to use for naming repository configuration secrets
+	repoConfigSecretPrefix = "repoconfig"
```

Comment:
This is to keep consistency with the value of the label, that is used to identify such secrets. For me, it would also be confusing to have to secrets with the same naming pattern, that are both managed by Argo CD, but have to slightly different semantics.

IMO the pre-existing prefix for credential secrets is rather unfortuned and we would be better off, if it would be something diffrent, eventhough I fail to come up with a better name, as `repcreds` is already taken 😉.

This is not a hill I want to die on, though. If you would prefere to re-use the name prefix here, I will make the change 🙂.

Reply:
I agree that it would be nice to have an easy way to distinguish between old style and new style secrets. My concern is that someone might think that we are introducing a new repo credential type and we will have to explain the difference between `repo` and `repoconfig`. 

What do you think about keeping old prefix and use function similar to `URIToSecretName` that would generate `repo-<hostname>_<underscore-delimited-paths>-<hash>` . In this case new repo secrets names will be visually different and also will include Git server host.

Reply:
I like your suggestion on theory, but I am unsure how to implement such a function, while guranteeing compliance with [Kubernetes name constraints](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/). It is certainly possible, but there are quit a few edge cases that need to be considered.

I will go with the `repo` prefix on this one, if you are okay with it?

Reply:
Sounds good !